firebase-firestore/ktx/src/main/kotlin/com/google/firebase/firestore/ktx/serialization/MapEncoderKtxImp.kt

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 Google LLC
```

Comment:
Why do you need this file and `FirestoreMapEconder` to exist separately?

Reply:
this `MapEncoderKtxImp` is also going to have a `decode()` method that will need the `FirestoreMapDecoder` class, and `decodeFromMap` inline function, so I am separating these three.

Alternatively, I can have two Java interface, `MapEncoder`, `MapDecoder`, and make the  `FirestoreMapEconder` and  `FirestoreMapDecoder` implement each of them separately.  Which method you think is better?